mesonbuild/mesonlib.py

```diff
@@ -1332,57 +1409,63 @@ def detect_subprojects(spdir_name, current_dir='', result=None):
                 result[basename] = [trial]
     return result
 
+
 def substring_is_in_list(substr: str, strlist: T.List[str]) -> bool:
     for s in strlist:
         if substr in s:
             return True
     return False
 
-class OrderedSet(collections.abc.MutableSet):
+
+class OrderedSet(T.MutableSet[_T]):
     """A set that preserves the order in which items are added, by first
     insertion.
     """
-    def __init__(self, iterable=None):
-        self.__container = collections.OrderedDict()
+    def __init__(self, iterable: T.Optional[T.Iterable[_T]] = None):
+        # typing.OrderedDict is new in 3.7.2, so we can't use that, but we can
```

Comment:
Since you will only be using `typing.OrderedDict` during type checking, you *can* actually use it if you want, since it is defined in the stubs that mypy uses. Understandable if you don't want to though.

Reply:
We run mypy with python 3.5 though, and it gets grumpy about undefined type, so I'm going to just use MutableMapping. Since the actual OrderedDict is a private implementation detail anyway.